Add sengled water leak sensor device

anzhuge opened the pull request:
1. Modify fingerprints.yml file for sengled water leak sensor.
2. Add the sendled directory under the src directory and add init.lua file under sendled directory.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SmartThingsCommunity/SmartThingsEdgeDrivers?pullRequest=615) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SmartThingsCommunity/SmartThingsEdgeDrivers?pullRequest=615) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SmartThingsCommunity/SmartThingsEdgeDrivers?pullRequest=615) it.</sub>

lelandblue commented:
Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/374

anzhuge commented:
Hello ***@***.***

As per your request, I have made new modifications and resubmitted the PR at: https://github.com/anzhuge/SmartThingsEdgeDrivers/pull/1




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.










At 2023-04-20 00:09:39, "lelandblue" ***@***.***> wrote:

Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: #374

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

anzhuge commented:
I'm sorry, please ignore the content.
















At 2023-04-23 19:30:40, "朱葛" ***@***.***> wrote:

Hello ***@***.***

As per your request, I have made new modifications and resubmitted the PR at: https://github.com/anzhuge/SmartThingsEdgeDrivers/pull/1




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.










At 2023-04-20 00:09:39, "lelandblue" ***@***.***> wrote:

Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: #374

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

anzhuge commented:
Hello ***@***.***

As per your request, I have made new modifications and committed.




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.













在 2023-04-23 19:50:19，"朱葛" ***@***.***> 写道：

I'm sorry, please ignore the content.
















At 2023-04-23 19:30:40, "朱葛" ***@***.***> wrote:

Hello ***@***.***

As per your request, I have made new modifications and resubmitted the PR at: https://github.com/anzhuge/SmartThingsEdgeDrivers/pull/1




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.










At 2023-04-20 00:09:39, "lelandblue" ***@***.***> wrote:

Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: #374

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

anzhuge commented:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.

anzhuge commented:
Hello, @lelandblue,

ctowns commented on drivers/SmartThings/zigbee-water-leak-sensor/src/sengled/init.lua:
```diff
@@ -0,0 +1,56 @@
+end
+
+local sengled_water_leak_sensor_handler = {
+  NAME = "Sengled Water Leak Sensor  Handler",
```
nit:
```suggestion
  NAME = "Sengled Water Leak Sensor Handler",
```

ctowns commented on drivers/SmartThings/zigbee-water-leak-sensor/fingerprints.yml:
```diff
@@ -99,3 +99,8 @@ zigbeeManufacturer:
     deviceProfileName: water-battery
+  - id: "Sengled E1L-G7K"
+    deviceLabel: Sengled Water Leak Sensor
+    manufacturer: sengled.
```
I believe the manufacturer name should just be "sengled"
```suggestion
    manufacturer: sengled
```

ctowns reviewed (commented):
Also, please squash your commits before merging

anzhuge commented:
Add Sengled rgbw bulb device
Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\src\rgbw-bulb\init.lua file, a new device type E23-N13 is added.
Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\fingerprints.yml file, a new device type E23-N13 is added.

anzhuge commented:
Add Sengled dimming light bulb device:
1. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\src\zigbee-dimming-light\init.lua file, a new device type E23-N11 is added.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\fingerprints.yml file, a new device type E23-N11 is added.

lelandblue commented:
@ctowns / @greens Can you please review this?

greens reviewed (changes requested):
sub-drivers must be `require`d in the base driver in order to be loaded and receive commands

greens commented:
conflicts need to be addressed with a rebase of the fork or the branch to origin/main

anzhuge commented:
> conflicts need to be addressed with a rebase of the fork or the branch to origin/main

I updated a color light and a white light on PR 615#. Can this be allowed?
PR 615# is my previous PR update on water leak sensor device.
